src/python/WMComponent/DBS3Buffer/DBSBufferBlock.py

```diff
@@ -294,20 +287,7 @@ def setDataset(self, datasetName, primaryType,
 
         Lexicon.primaryDatasetType(primaryType)
 
-        if not datasetType in ['VALID', 'PRODUCTION', 'INVALID', 'DEPRECATED', 'DELETED']:
-            msg = "Invalid processedDatasetType %s\n" % datasetType
-            logging.error(msg)
-            raise DBSBufferBlockException(msg)
-
-        try:
-            if datasetName[0] == '/':
-                _, primary, processed, tier = datasetName.split('/')
-            else:
-                primary, processed, tier = datasetName.split('/')
-        except Exception:
-            msg = "Invalid dataset name %s" % datasetName
-            logging.error(msg)
-            raise DBSBufferBlockException(msg)
+        _, primary, processed, tier = datasetName.split('/')
```

Comment:
Isn't the check for leading `/` still needed here, even though we remove the `datasetType` check?

Reply:
It shouldn't, since it's a dataset name. If it's broken, then there is a serious error upstream and it will fail as we try to inject it into DBS. So, I don't think we gain anything from that extra check.